### PR TITLE
User input

### LINKED ISSUE #10: Mention input events for display locking
We need to outline what happens with input events when an element is locked for display. Ignore all of them?

### PR DESCRIPTION
**Pull request by vmpstr**
Closes #10

### CODE REVIEW

File: display-locking/explainer.md [hunk 1]
```diff
@@ -467,7 +467,49 @@ This allows us to better reason about the expected behavior of the page. Specifi
   changing the content on the rest of the page.
 
 ---
-### Edge cases
+### Dealing with user input
+
+One of the difficult aspects of locking an element for display is deciding what
+to do with user input (e.g. mouse clicks) that are targetted at the element
```
Comment (chrishtr) [inline display-locking/explainer.md]: s/targetted/targeted/

File: display-locking/explainer.md [hunk 2]
```diff
@@ -467,7 +467,49 @@ This allows us to better reason about the expected behavior of the page. Specifi
...
+which is locked for display.
+
+##### Queuing up input and replaying
+
+One option is to queue up the user input and replay it once the element is
```
Comment (chrishtr) [inline display-locking/explainer.md]: s/replay/replaying/

File: display-locking/explainer.md [hunk 3]
```diff
@@ -467,7 +467,49 @@ This allows us to better reason about the expected behavior of the page. Specifi
...
+unlocked.
+
+Immediately, there is a problem with this approach: conceptually, user
+input is targetted at what the user is currently seeing on screen. However, when
```
Comment (chrishtr) [inline display-locking/explainer.md]: s/targetted/targeted/

**Comment by vmpstr** [comment]
Addressed comments, PTAL.

File: display-locking/explainer.md [hunk 4]
```diff
@@ -467,7 +467,49 @@ This allows us to better reason about the expected behavior of the page. Specifi
...
+One option is to queue up the user input replaying it once the element is
+unlocked.
+
+Immediately, there is a problem with this approach: conceptually, user
+input is targeted at what the user is currently seeing on screen. However, when
+the display is locked for view, it means that whatever is currently visible does
+not necessarily reflect the DOM that currently comprises the element. To put it
+differently, script could have already mutated the DOM, removed interactive
+elements, added more interactive elements, etc, which means that the user input
+may be targetting an unexpected (from the user's perspective) element.
```
Comment (chrishtr) [inline display-locking/explainer.md]: s/targetting/targeting/

**Pull request merged**